test/plugin/test_out_detect_exceptions.rb

```diff
@@ -98,6 +114,80 @@ def test_exception_detection
     assert_equal(make_logs(t, *messages), d.events)
   end
 
+  def test_ignore_exception_nested_in_json
```

Comment:
I'd prefer to keep them in separate test cases, because that way we know exactly what expectation failed. However, there's so much common code that refactoring makes sense.